docarray/index/abstract.py

```diff
@@ -706,6 +707,21 @@ def _filter_by_parent_id(self, id: str) -> Optional[List[str]]:
     # These might be useful in your subclass implementation  #
     ##########################################################
 
+    @staticmethod
+    def custom_issubclass(x, A_tuple) -> bool:
```

Comment:
Let's make this a function instead of a staticmethod, and let's more it to `utils._internal._typing`.
Maybe we can also rename it to `safe_issubclass` to indicate a bit more about what makes it custom?